packages/react-router/tests/link.test.tsx

```diff
@@ -619,6 +668,64 @@ describe('Link', () => {
     expect(await screen.findByText('UserId: userId')).toBeInTheDocument()
   })
 
+  test('when navigating to /posts with a beforeLoad that throws an error', async () => {
+    const onError = vi.fn()
+    const rootRoute = createRootRouteWithContext<{
```

Comment:
why do we need router context here?
If we don't need it for this test, please remove to simplify the test